go/libkb/upak.go

```diff
@@ -69,15 +69,15 @@ func CheckKID(u *keybase1.UserPlusKeysV2AllIncarnations, kid keybase1.KID) (foun
 	return checkKIDKeybase(u, kid)
 }
 
-func GetRemoteChainLinkFor(u *keybase1.UserPlusAllKeys, username NormalizedUsername, uid keybase1.UID, g *GlobalContext) (ret *TrackChainLink, err error) {
-	defer g.Trace(fmt.Sprintf("UPAK#GetRemoteChainLinkFor(%s,%s,%s)", u.Base.Uid, username, uid), func() error { return err })()
+func GetRemoteChainLinkFor(u *keybase1.UserPlusKeysV2AllIncarnations, username NormalizedUsername, uid keybase1.UID, g *GlobalContext) (ret *TrackChainLink, err error) {
```

Comment:
globalContext here is passed last, maybe update?

I suggest we throw an error here if username or uid are not being passed (i.e. they are nil or ""). This is because before we were returning nil,nil if uid was nil and now we do that if the username is nil....
In general, is it documented when it is ok to return nil as "all went well but not found" and when it isn't?